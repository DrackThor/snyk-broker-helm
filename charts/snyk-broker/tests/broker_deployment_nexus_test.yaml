suite: broker deployment (nexus)
chart:
  version: 0.0.0
values:
  - ./fixtures/default_values.yaml

tests:
  - it: should create secret if brokerClientValidationUrl, baseNexusUrl and nexusUrl are defined
    template: secrets.yaml

    set:
      scmType: nexus
      baseNexusUrl: https://username:password@your-domain.com
      nexusUrl: https://username:password@your-domain.com/repository
      brokerClientValidationUrl: https://username:password@your-domain.com/service/rest/v1/status/check
      disableSuffixes: true
    documentIndex: 3

    asserts:
      - containsDocument:
          apiVersion: v1
          kind: Secret
          name: nexus-broker-client-validation-url
      - equal:
          path: stringData.nexus-broker-client-validation-url
          value: https://username:password@your-domain.com/service/rest/v1/status/check

  - it: should create secret if brokerClientValidationUrl and nexusUrl are defined
    template: secrets.yaml

    set:
      scmType: nexus
      nexusUrl: https://username:password@your-domain.com/repository
      brokerClientValidationUrl: https://username:password@your-domain.com/service/rest/v1/status/check
      disableSuffixes: true
    documentIndex: 2

    asserts:
      - containsDocument:
          apiVersion: v1
          kind: Secret
          name: nexus-broker-client-validation-url
      - equal:
          path: stringData.nexus-broker-client-validation-url
          value: https://username:password@your-domain.com/service/rest/v1/status/check

  - it: should create secret if brokerClientValidationUrl and baseNexusUrl are defined
    template: secrets.yaml

    set:
      scmType: nexus
      baseNexusUrl: https://username:password@your-domain.com
      brokerClientValidationUrl: https://username:password@your-domain.com/service/rest/v1/status/check
      disableSuffixes: true
    documentIndex: 2

    asserts:
      - containsDocument:
          apiVersion: v1
          kind: Secret
          name: nexus-broker-client-validation-url
      - equal:
          path: stringData.nexus-broker-client-validation-url
          value: https://username:password@your-domain.com/service/rest/v1/status/check

  - it: should not create secret for brokerClientValidationUrl if value is empty
    template: secrets.yaml

    set:
      scmType: nexus
      nexusUrl: https://username:password@your-domain.com/repository
      baseNexusUrl: https://username:password@your-domain.com
      disableSuffixes: true

    asserts:
      - notEqual:
          path: metadata.name
          value: nexus-broker-client-validation-url

  - it: should render render nexusUrl, baseNexusUrl and brokerClientValidationUrl as secrets
    templates:
      - broker_deployment.yaml
      - broker_service.yaml
      - secrets.yaml
      - serviceaccount.yaml

    set:
      scmType: nexus
      baseNexusUrl: https://username:password@your-domain.com
      nexusUrl: https://username:password@your-domain.com/repository
      brokerClientValidationUrl: https://username:password@your-domain.com/service/rest/v1/status/check

    asserts:
      - matchSnapshot: {}
