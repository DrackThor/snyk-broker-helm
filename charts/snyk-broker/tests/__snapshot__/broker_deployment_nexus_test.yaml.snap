should render render nexusUrl, baseNexusUrl and brokerClientValidationUrl as secrets:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: snyk-broker-RELEASE-NAME
        helm.sh/chart: snyk-broker-0.0.0
        node1: test1
        node2: test2
      name: nexus-broker-RELEASE-NAME
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: snyk-broker-RELEASE-NAME
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: snyk-broker-RELEASE-NAME
        spec:
          containers:
            - env:
                - name: BROKER_SERVER_URL
                  value: https://broker.test.snyk.io
                - name: BROKER_HEALTHCHECK_PATH
                  value: /healthcheck
                - name: BROKER_SYSTEMCHECK_PATH
                  value: /systemcheck
                - name: BROKER_TOKEN
                  valueFrom:
                    secretKeyRef:
                      key: nexus-broker-token-key
                      name: nexus-broker-token-RELEASE-NAME
                - name: BASE_NEXUS_URL
                  valueFrom:
                    secretKeyRef:
                      key: nexus-base-nexus-url
                      name: nexus-base-nexus-url-RELEASE-NAME
                - name: NEXUS_URL
                  valueFrom:
                    secretKeyRef:
                      key: nexus-nexus-url
                      name: nexus-nexus-url-RELEASE-NAME
                - name: BROKER_CLIENT_VALIDATION_URL
                  valueFrom:
                    secretKeyRef:
                      key: nexus-broker-client-validation-url
                      name: nexus-broker-client-validation-url-RELEASE-NAME
                - name: PORT
                  value: "8000"
                - name: LOG_LEVEL
                  value: info
                - name: LOG_ENABLE_BODY
                  value: "false"
                - name: BROKER_DISPATCHER_BASE_URL
                  value: https://api.test.snyk.io
              image: snyk/broker:nexus
              imagePullPolicy: Always
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              name: nexus-broker-RELEASE-NAME
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /healthcheck
                  port: 8000
                  scheme: HTTP
                initialDelaySeconds: 3
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 1
                  memory: 256Mi
                requests:
                  cpu: 1
                  memory: 256Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
              volumeMounts: null
          securityContext: {}
          serviceAccountName: snyk-broker-RELEASE-NAME
          volumes: null
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: snyk-broker-RELEASE-NAME
        helm.sh/chart: snyk-broker-0.0.0
        node1: test1
        node2: test2
      name: nexus-broker-service-RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8000
          targetPort: 8000
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: snyk-broker-RELEASE-NAME
      type: ClusterIP
  3: |
    apiVersion: v1
    data:
      nexus-broker-token-key: MTIz
    kind: Secret
    metadata:
      name: nexus-broker-token-RELEASE-NAME
    type: Opaque
  4: |
    apiVersion: v1
    data:
      nexus-base-nexus-url: aHR0cHM6Ly91c2VybmFtZTpwYXNzd29yZEB5b3VyLWRvbWFpbi5jb20=
    kind: Secret
    metadata:
      name: nexus-base-nexus-url-RELEASE-NAME
    type: Opaque
  5: |
    apiVersion: v1
    data:
      nexus-nexus-url: aHR0cHM6Ly91c2VybmFtZTpwYXNzd29yZEB5b3VyLWRvbWFpbi5jb20vcmVwb3NpdG9yeQ==
    kind: Secret
    metadata:
      name: nexus-nexus-url-RELEASE-NAME
    type: Opaque
  6: |
    apiVersion: v1
    kind: Secret
    metadata:
      name: nexus-broker-client-validation-url-RELEASE-NAME
    stringData:
      nexus-broker-client-validation-url: https://username:password@your-domain.com/service/rest/v1/status/check
    type: Opaque
  7: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: snyk-broker-RELEASE-NAME
        helm.sh/chart: snyk-broker-0.0.0
        node1: test1
        node2: test2
      name: snyk-broker-RELEASE-NAME
      namespace: NAMESPACE
